app/models/manageiq/providers/amazon/cloud_manager.rb

```diff
@@ -152,13 +166,26 @@ def self.params_for_create
         },
         {
           :component => "text-field",
-          :name      => "endpoints.default.assume_role",
+          :name      => "endpoints.default.service_account",
           :label     => "Assume Role"
         }
       ]
     }.freeze
   end
 
+  def self.create_from_params(params)
```

Comment:
Would be nice if this method would exist on the top-level `Provider` class and it would be able to select the right one based on the `type` attribute of the passed `params`. This way we would not need any provider-specific logic in the API.

Reply:
But that's not in the context of this PR

Reply:
Yeah I'm playing around with maybe this can be completely generic if we setup the params properly, or at least do the bulk of it generically and the providers only override anything weird that they do